### PR TITLE
Implement menu button for clear verification

### PR DESCRIPTION
**Pull request by OTCompa**
Dependent on #32 
Implements a modal when the `Verify Clears` button is clicked that verifies clears upon submission.
I've also separated the `Clears()` function into another helper so I can reuse the code.

image
image
image
image
(Used an alt because my main would fill the entire screen)

### CODE REVIEW

File: internal/clearingway/handlers.go [hunk 1]
```diff
@@ -211,10 +211,10 @@ var ReclearCommand = &discordgo.ApplicationCommand{
...
 				// TODO: implement when FRU goes live
 				/*
-				{
-					Name: "FRU",
-					Value: "Futures Rewritten (Ultimate)",
-				},
+					{
+						Name: "FRU",
+						Value: "Futures Rewritten (Ultimate)",
+					},
```
Comment (brtran4) [inline internal/clearingway/handlers.go]: fmt thing showed up again

File: internal/clearingway/menu.go [hunk 2]
```diff
@@ -192,3 +193,89 @@ func (ms *Menus) Roles() *Roles {
...
+			Components: []discordgo.MessageComponent{
+				discordgo.ActionsRow{
+					Components: []discordgo.MessageComponent{
+						discordgo.TextInput{
+							CustomID:    "firstName",
+							Style:       discordgo.TextInputShort,
+							Label:       "Character first name",
+							Placeholder: "First name",
+							Required:    true,
+							MaxLength:   15,
```
Comment (brtran4) [inline internal/clearingway/menu.go:215]: is this verified to be the longest string possible for a name?

File: internal/clearingway/menu.go [hunk 3]
```diff
@@ -192,3 +193,89 @@ func (ms *Menus) Roles() *Roles {
...
+				discordgo.ActionsRow{
+					Components: []discordgo.MessageComponent{
+						discordgo.TextInput{
+							CustomID:    "world",
+							Style:       discordgo.TextInputShort,
+							Label:       "Character world",
+							Placeholder: "World",
+							Required:    true,
+							MaxLength:   20,
+							MinLength:   2,
```
Comment (brtran4) [inline internal/clearingway/menu.go:242]: is it possible to have an auto-complete option for the worlds? there seems to be a feature in the current implementation of clearingway 
image

File: internal/clearingway/menu.go (hunk 2 above)
Comment (OTCompa) [reply internal/clearingway/menu.go:215]: yup!
image

File: internal/clearingway/menu.go (hunk 3 above)
Comment (OTCompa) [reply internal/clearingway/menu.go:242]: I don't believe Discord modals support autocomplete (or any form of input other than a text box)

File: internal/clearingway/handlers.go (hunk 1 above)
Comment (OTCompa) [reply internal/clearingway/handlers.go]: oops, gofmt automatically formats this so i have to remind myself whenever I save in handlers.go

**Review by brtran4 — COMMENTED** [review]
github is being dumb and wont let me comment on the line 🙃 but we should double check that `cleanWorld` in `clears-handler.go` is still parsing the world correctly without the autofill (such as typing the world with lowercase). i might be too paranoid about this but i dont really trust the user to actually type the world in right

image

**Review by brtran4 — APPROVED** [review]

**Comment by OTCompa** [comment]
> github is being dumb and wont let me comment on the line 🙃 but we should double check that `cleanWorld` in `clears-handler.go` is still parsing the world correctly without the autofill (such as typing the world with lowercase). i might be too paranoid about this but i dont really trust the user to actually type the world in right
> 
> image

doesn't seem like there's any issue with lowercase worlds!
https://github.com/user-attachments/assets/bc2779d1-337c-47ed-b9e7-f8fcc05ad691

**Comment by brtran4** [comment]
> > github is being dumb and wont let me comment on the line 🙃 but we should double check that `cleanWorld` in `clears-handler.go` is still parsing the world correctly without the autofill (such as typing the world with lowercase). i might be too paranoid about this but i dont really trust the user to actually type the world in right
> > image
> 
> doesn't seem like there's any issue with lowercase worlds! https://github.com/user-attachments/assets/bc2779d1-337c-47ed-b9e7-f8fcc05ad691

amazing lgtm

**Pull request merged**